src/Compilers/Core/CodeAnalysisTest/Text/StringBuilderTextTest.cs

```diff
@@ -0,0 +1,52 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Text;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.UnitTests.Text
+{
+    public sealed class StringBuilderTextTest
+    {
+        [Fact]
+        public void Write()
+        {
+            var stringBuilder = new StringBuilder(6);
+
+            // This way, the StringBuilder should have at least two chunks of length 6 at the beginning. We want to test writes
+            // across their boundaries.
+            stringBuilder.Append("chunk1");
+            stringBuilder.Append("chunk2");
+            stringBuilder.Append("chunk3");
```

Comment:
Could we add an `Assert` here to verify that this produces at least two chunks in the data? The internal structure of chunks seems subject to change so would like some verification we test this code path over the long term.

Reply:
Done.